### PR TITLE
Improved, per-persona validation messages

### LINKED ISSUE #97: Consistent & Clear Wording in Validations
**Extended Description**

As a FedRAMP reviewer, to best understand if the validations I and CSP SSP authors will read actually reflect the intended guidance from a formal FedRAMP review, I want to ensure the current validation output messages are consistent and clear in how they communicate errors and important diagnostic information about the SSP.

**Preconditions**
- [ ] Access to example review reports (with SSP feedback) for one or more real FedRAMP CSP packages
- [ ] Complete this work, if possible, alongside 18F/fedramp-automation#166.

**Acceptance Criteria**
- [ ] For high-level validations that map to checklist items match wording as best as possible with real-world examples provided (terminology, phrasing, etc.)
- [ ] Create a shared document with a glossary so we can align on terms with the FART: add words or phrases we will define to ask them for periodic review

**Story Tasks**
- [ ] Review User Feedback and add tasks for specific items to fix
- [ ] For the reviewer persona, update the following wording in the messaging (`@id` can stay the same):
  - [ ] Generally for rule assertions on lead-in since we are talking prescriptive rules, change `This SSP has` to `A FedRAMP SSP must` to make it clear the aspirational nature of the rule.
  - [ ] `base64` -> `attachment file content`
  - [ ] `resource` -> `supporting artifact found in a citation`
  - [ ] `A FedRAMP SSP has defined a responsible party with no extraneous roles.` -> `Responsible Role in control does not correspond to any entry in Roles and Responsibilities Table.`
  - [ ] `A FedRAMP SSP must not include implemented controls beyond what is required for the applied baseline. ... A FedRAMP SSP has implemented X extraneous controls not needed given the selected profile`.

**More Examples**
The following adopt a "business rule" style.
```
A FedRAMP SSP submission must include a Security Assessment Plan (SAP).
A FedRAMP SSP submission must include a Security Assessment Report (SAR).
A FedRAMP SSP submission must include Plans of Action and Milestones (POA&Ms).
A FedRAMP SSP submission must include a Continuous Monitoring Plan (ConMon Plan).
A FedRAMP SSP must use the correct FedRAMP SSP Template.
A FedRAMP SSP must use the correct FedRAMP Deployment Model.
Within a FedRAMP SSP, all controls have at least one implementation status checkbox selected.
Within a FedRAMP SSP, all critical controls are implemented.
Within a FedRAMP SSP, customer responsibilities are clearly identified in the CIS-CRM Tab, as well as the SSP Controls (by checkbox selected and in the implementation description). The CIS-CRM and SSP controls are consistent for customer responsibilities.
Within a FedRAMP SSP, the Initial Authorizing Agency concurs with the CRM (adequacy and clarity of customer responsibilities).
Within a FedRAMP SSP, the Roles Table (User Roles and Privileges) sufficiently describes the range of user roles, responsibilities, and access privileges.
Within a FedRAMP SSP, in the control summary tables, the information in the Responsible Role row correctly describes the required entities responsible for fulfilling the control.
Within a FedRAMP SSP, the appropriate Digital Identity Level is selected.
Within a FedRAMP SSP, the authorization boundary is explicitly identified in the network diagram.
Within a FedRAMP SSP, does the CSO provide components to run on the client side?
Within a FedRAMP SSP, there is a data flow diagram that clearly illustrates the flow and protection of data going in and out of the service boundary and that includes all traffic flows for both internal and external users.
Within a FedRAMP SSP, does the CSP use any third-party or external cloud services that lack FedRAMP Authorization?
Within a FedRAMP SSP, if this is a SaaS or a PaaS, is it "leveraging" another IaaS with a FedRAMP Authorization?
Within a FedRAMP SSP, if 11a is Yes, the "inherited" controls are clearly identified in the control descriptions.
Within a FedRAMP SSP, all interconnections are correctly identified and documented in the SSP.
Within a FedRAMP SSP, all required controls are present.
Within a FedRAMP SSP, the inventory is provided in the FedRAMP Integrated Inventory Workbook.
Within a FedRAMP SSP, the CSO is compliant with DNSSEC. (Controls SC-20 and SC-21 apply).
Within a FedRAMP SSP, the CSO adequately employs Domain-based Message Authentication, Reporting & Conformance (DMARC) requirements according to DHS BOD 18-01.
A FedRAMP SSP must incorporate the FedRAMP Master Acronym and Glossary.
A FedRAMP SSP must incorporate the FedRAMP Applicable Laws and Regulations.
A FedRAMP SSP must incorporate the FedRAMP Logo.
A FedRAMP SSP must incorporate a User Guide.
A FedRAMP SSP must incorporate Rules of Behavior.
A FedRAMP SSP must incorporate a Contingency Plan.
A FedRAMP SSP must incorporate a Configuration Management Plan.
A FedRAMP SSP must incorporate an Incident Response Plan.
A FedRAMP SSP must incorporate a Separation of Duties Matrix.
A FedRAMP SSP must incorporate a policy document for each of the 17 NIST SP 800-54 Revision 4 control families.
A FedRAMP SSP must incorporate a procedure document for each of the 17 NIST SP 800-54 Revision 4 control families.
FedRAMP SSP policy and procedure documents must have unique per-control-family associations.
A FedRAMP SSP must incorporate a Privacy Point of Contact role.
A FedRAMP SSP must have all four PTA questions answered.
A FedRAMP SSP must incorporate one or more NIST CMVP-validated cryptographic modules (FIPS 140).
A FedRAMP SSP must specify a FIPS 199 categorization.
A FedRAMP SSP must specify one or more SP 800-60v2r1 information types.
A FedRAMP SSP must incorporate a Digital Identity Determination.
A FedRAMP SSP must incorporate a system inventory.
A FedRAMP SSP must have a FedRAMP system-id.
A FedRAMP SSP must have a system name.
A FedRAMP SSP must have a short system name.
A FedRAMP SSP must have a FedRAMP authorization type.
A FedRAMP SSP must identify the system owner.
A FedRAMP SSP must identify the authorizing official.
A FedRAMP SSP must identify the system management point of contact.
A FedRAMP SSP must identify the system technical point of contact.
A FedRAMP SSP must identify the system other point of contact.
A FedRAMP SSP must incorporate an authorization boundary diagram.
A FedRAMP SSP must incorporate a network-architecture diagram.
A FedRAMP SSP must incorporate a data-flow diagram.
A FedRAMP SSP employs a FedRAMP OSCAL profile.
Within a FedRAMP SSP, every required control must have an implementation status.
Within a FedRAMP SSP, planned control implementations must have a planned completion date.
```

**Definition of Done**
 - [ ] Acceptance criteria met - Each user story should meet the acceptance criteria in the description
 - [ ] Unit test coverage of our code > 90%  (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*) this may be fuzzy and hard to prove
 - [ ] Code quality checks passed - Enable html tidy with XML code standards as part of the build (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*)
~~- [ ] Accessibility: (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*) as we create guidance or documentation and reports (semantic tagging including aria tags): demonstrate with 0 errors reported for WCAG 2.1 AA standards using an automated scanner and 0 errors reported in manual testing~~
 - [ ] Code reviewed - Code reviewed by at least one other team members (or developed by a pair)
 - [ ] Source code merged - Code that’s demoed must be in source control and merged
~~- [ ] Code must successfully build and deploy into staging environment (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*): this may evolve from xslt sh pipline into something more~~
~~- [ ] Security reviewed and reported - Conduct vulnerability and compliance scanning. threat modeling? ~~
~~- [ ] Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*)~~
~~- [ ] Usability tests passed - Each user story should be easy to use by target users (development community? FedRAMP FART team)~~
~~- [ ] Usability testing and other user research methods must be conducted at regular intervals throughout the development process (not just at the beginning or end). (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*)~~
 - [ ] Code refactored for clarity - Code must be clean, self-documenting
 - [ ] No local design debt
 - [ ] Load/performance tests passed - test data needed - saxon instrumentation
 - [ ] Documentation generated - update readme or contributing markdown as necessary.
~~- [ ] Architectural Decision Record completed as necessary for significant design choices~~

### LINKED ISSUE #189: Customize Validation Messages by Submitter & Approver
**Extended Description**
As a Submitter & an Approver, in order to view the most accurate feedback using language and terms most familiar with, I would like the validation messages to be geared towards my role as a Submitter or Approver. 

**Preconditions**
- Preconditions...

**Acceptance Criteria**
- Approver messages are sent to a subset of Approvers to review.
- Submitter messages are sent to a subset of Submitters for review.

**Story Tasks**
- [ ] Tasks...

**Definition of Done**
 - [ ] Acceptance criteria met - Each user story should meet the acceptance criteria in the description
 - [ ] Unit test coverage of our code > 90%  (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*) this may be fuzzy and hard to prove
 - [ ] Code quality checks passed - Enable html tidy with XML code standards as part of the build (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*)
 - [ ] Accessibility: (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*) as we create guidance or documentation and reports (semantic tagging including aria tags): demonstrate with 0 errors reported for WCAG 2.1 AA standards using an automated scanner and 0 errors reported in manual testing
 - [ ] Code reviewed - Code reviewed by at least one other team members (or developed by a pair)
 - [ ] Source code merged - Code that’s demoed must be in source control and merged
 - [ ] Code must successfully build and deploy into staging environment (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*): this may evolve from xslt sh pipline into something more
 - [ ] Security reviewed and reported - Conduct vulnerability and compliance scanning. threat modeling? 
 - [ ] Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*)
 - [ ] Usability tests passed - Each user story should be easy to use by target users (development community? FedRAMP FART team)
 - [ ] Usability testing and other user research methods must be conducted at regular intervals throughout the development process (not just at the beginning or end). (*from [QASP](https://github.com/flexion/TTS-10x/wiki/TTS-QASP)*)
 - [ ] Code refactored for clarity - Code must be clean, self-documenting
 - [ ] No local design debt
 - [ ] Load/performance tests passed - test data needed - saxon instrumentation
 - [ ] Documentation generated - update readme or contributing markdown as necessary.
 - [ ] Architectural Decision Record completed as necessary for significant design choices

### PR DESCRIPTION
**Pull request by GaryGapinski**
This PR combines changes for #97 and #189.

The Schematron validation rules now contain per-persona message variants.
These are segregated in a "foreign" XML namespace and are indexed by Schematron assertion ID, persona ID, and affirmative|diagnostic message.

**NB**: Neither affirmative nor diagnostic person-specific messages which use run-time text replacements (`<sch:name>`, `<sch:value-of>`) can be employed for SVRL generation unless the Schematron document is transformed by transferring the desired persona-specific messages to the Schematron assertions prior to validation.

Once the per-persona message content is deemed of good quality and pertinent to the venue, the "reviewer" variant can replace the Schematron messages (under the assumption "reviewer" is the likeliest near-term venue) using a transform as described above.

This PR will close #97.
This PR will close #189.
Supports #166.

### CODE REVIEW

File: src/validations/rules/ssp.sch [hunk 1]
```diff
@@ -520,7 +520,7 @@
                 doc:organizational-id="section-b.?????"
                 id="resource-rlink-required"
                 role="error"
-                test="doc-available(./@href)">This SSP references back-matter resource: <sch:value-of
+                test="doc-available(./@href)">A FedRAMP SSP must references back-matter resource: <sch:value-of
```
Comment (danielnaab) [inline src/validations/rules/ssp.sch:523]:
```suggestion
                test="doc-available(./@href)">A FedRAMP SSP must reference back-matter resource: <sch:value-of
```

File: src/validations/rules/ssp.sch [hunk 2]
```diff
@@ -2330,10 +2330,10 @@ leveraged-authorization.</sch:assert>
         <sch:value-of select="$extraneous-parties/o:party-uuid" />.</sch:diagnostic>
         <sch:diagnostic doc:assertion="resource-uuid-required"
                         doc:context="/o:system-security-plan/o:back-matter/o:resource"
-                        id="resource-uuid-required-diagnostic">This SSP includes back-matter resource missing a UUID.</sch:diagnostic>
+                        id="resource-uuid-required-diagnostic">A FedRAMP SSP must include back-matter resource missing a UUID.</sch:diagnostic>
```
Comment (danielnaab) [inline src/validations/rules/ssp.sch:2333]:
```suggestion
                        id="resource-uuid-required-diagnostic">A FedRAMP SSP back-matter resource must include a UUID.</sch:diagnostic>
```

File: src/validations/rules/ssp.sch [hunk 3]
```diff
@@ -2874,7 +2874,7 @@ leveraged-authorization.</sch:assert>
                         id="user-has-authorized-privilege-diagnostic">This user lacks one or more authorized-privileges.</sch:diagnostic>
         <sch:diagnostic doc:assertion="user-user-type-has-allowed-value"
                         doc:context="oscal:user/oscal:prop[@name = 'type']"
-                        id="user-user-type-has-allowed-value-diagnostic">User type property has an allowed value.</sch:diagnostic>
+                        id="user-user-type-has-allowed-value-diagnostic">This user type property lacks an allowed value.</sch:diagnostic>
```
Comment (danielnaab) [inline src/validations/rules/ssp.sch:2877]: The framing is different here than the other diagnostic changes - should it be along the lines of "user type properties must only include allowed values"?

**Review by danielnaab — COMMENTED** [review]
Interesting approach on the method of declaring per-persona messages - I will think though how we might work with that in the UI.

I was confused on some of the tense changes on diagnostic messages - with the changes, were they all to be statements of something lacking, or switch to a statement of the rule that was violated?

File: src/validations/rules/ssp.sch (hunk 3 above)
Comment (GaryGapinski) [reply src/validations/rules/ssp.sch:2877]: Multiple diagnostic messages (the ones accompanying failed assertions) were not appropriate (they essentially duplicated the affirmative message). The assertion message should be an affirmative one (successful test being an affirmation of the messages) and a failed assertion message (diagnostic) should be a description of the deficiency in the context of the rule.

Comment (GaryGapinski) [reply src/validations/rules/ssp.sch:2877]: This was somewhat related to https://github.com/18F/fedramp-automation/issues/97.

**Review by danielnaab — APPROVED** [review]

**Review by ohsh6o — CHANGES_REQUESTED** [review]
Prelim review from yesterday, sorry I did not submit it until now. Still going through it more thoroughly now.

File: src/validations/rules/ssp.sch [hunk 4]
```diff
@@ -676,57 +657,51 @@
...
                         id="has-configuration-management-plan"
                         role="error"
                         test="oscal:resource[oscal:prop[@ns = 'https://fedramp.gov/ns/oscal' and @name = 'type' and @value = 'configuration-management-plan']]">
                         [Section B Check 3.7] A FedRAMP OSCAL SSP must have a Configuration Management Plan attached.</sch:assert>
             <sch:assert diagnostics="has-incident-response-plan-diagnostic"
-                        doc:attachment="§15 Attachment 8"
                         doc:checklist-reference="Section B Check 3.8"
                         doc:guide-reference="DRAFT Guide to OSCAL-based FedRAMP System Security Plans §6"
-                        doc:template-reference="System Security Plan Template §15"
+                        doc:template-reference="System Security Plan Template §155 Attachment 8"
```
[the remark this answers is not in the document]
Comment (ohsh6o) [reply src/validations/rules/ssp.sch]: I will close this one. This typo is resolved already.

File: src/validations/rules/ssp.sch [hunk 5]
```diff
@@ -271,14 +271,12 @@
             <sch:assert diagnostics="no-security-sensitivity-level-diagnostic"
                         doc:checklist-reference="Section C Check 1.a"
                         doc:guide-reference="DRAFT Guide to OSCAL-based FedRAMP System Security Plans §4.2"
-                        doc:organizational-id="section-c.1.a"
```
[the remark this answers is not in the document]
Comment (GaryGapinski) [reply src/validations/rules/ssp.sch:274]: organizational-id has been erased.

**Pull request merged**